src/config/configuration_loader.coffee

```diff
@@ -38,11 +37,16 @@ class ConfigurationLoader
 
 
   loadDefaultConfig: (done) =>
+    fsExtra = require 'fs-extra'
     fsExtra.readJson @defaultConfigPath, done
 
 
   loadUserConfig: (done) =>
-    filePaths = _.map extensions, (ext) => path.join @dir, "dependency-lint.#{ext}"
+    async = require 'async'
```

Comment:
I'm fine pushing out anything required more than once into an outer scope to keep things dry.

I expect reviews to be able to catch your last point. And there's nothing stopping that from happening when all the requires were at the top of the file